source/lib/login.js

```diff
@@ -45,10 +46,16 @@ export async function performLogin(
   }
 
   const form = buildFormData({username, password})
+  let loginFailed = false
   const loginResult = await fetch(OLECARD_AUTH_URL, {
     method: 'POST',
     body: form,
+  }).catch(() => {
+    loginFailed = true
   })
+  if (loginFailed) {
+    return false
+  }
```

Comment:
Could you change this to something like 

```js
    const form = buildFormData({username, password})
    
    let loginResult = null
    try {
      loginResult = await fetch(OLECARD_AUTH_URL, {
        method: 'POST',
        body: form,
      })
    } catch (err) {
      return false
    }

    if (!loginResult) {
      return false
    }

    const page = await loginResult.text()

    if (page.includes('Password')) {
```

I don't like to assign to things outside of a Promise from a promise callback … it feels fragile. Instead, I used a normal try/catch block around the `await` keyword to catch the network error, and return false from the catch block.